packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1407,6 +1410,7 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
         if (prefixIcon != null) {
           start -= contentPadding.start;
           start += centerLayout(prefixIcon!, start);
+          start += prefixToInputGap;
```

Comment:
This change is for the `TextDirection.ltr` case, I would also expect a similar change for the suffix (probably at line 1438)